OpenRA.Mods.Common/Installer/ISourceAction.cs

```diff
@@ -0,0 +1,21 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.Installer
+{
+	public interface ISourceAction
+	{
+		void Run(MiniYaml actionYaml, string path, ModData modData, List<string> extracted, Action<string> updateMessage);
```

Comment:
This name is too generic IMO. Can we call it something that better describes what it does?